### DISCUSSION TITLE
Property name expected type of string but got null

### DISCUSSION THREAD

**Discussion by PikabobAlex** [google/model-viewer#2556] [Q&A]
node_modules\@google\model-viewer\dist\model-viewer.min.js: Property name expected type of string but got null

hi guys, what does it mean?

I imported the js from like this
import modelViewer from "@google/model-viewer"

**Comment by shamsail** [comment]
Hi, If you have installed this package already , you can use it like this.

```
import { ModelViewerElement } from "@google/model-viewer/dist/model-viewer.js";
```

**Reply by PikabobAlex** [reply]
Hi, thanks for reply but it's still the same error. 

node_modules\@google\model-viewer\dist\model-viewer.js: Property name expected type of string but got null

**Reply by shamsail** [reply]
How you have installed this package? and where are you using this package? It will give help to understand what's going on at your side.

**Reply by burakkaraceylan** [reply]
I have the same problem. It works fine in development but when I try to build the app with parcel I get

×  E:\test\node_modules\@google\model-viewer\dist\model-viewer.js: E:\test\node_modules\@google\model-viewer\dist\model-viewer.js: Property name expected type of string but got null
    at validate (E:\test\node_modules\@babel\types\lib\definitions\utils.js:158:13)
    at Object.validate (E:\test\node_modules\@babel\types\lib\definitions\utils.js:227:7)
    at validateField (E:\test\node_modules\@babel\types\lib\validators\validate.js:24:9)
    at validate (E:\test\node_modules\@babel\types\lib\validators\validate.js:17:3)
    at builder (E:\test\node_modules\@babel\types\lib\builders\builder.js:38:27)
    at Object.identifier (E:\test\node_modules\@babel\types\lib\builders\generated\index.js:343:31)
    at E:\test\node_modules\regenerator-transform\lib\hoist.js:32:29
    at Array.forEach (<anonymous>)
    at varDeclToExpr (E:\test\node_modules\regenerator-transform\lib\hoist.js:29:23)
    at exit (E:\test\node_modules\regenerator-transform\lib\hoist.js:51:20)

**Reply by shamsail** [reply]
now I'm getting your problem. your problem is not with `model-viewer` library itself. your error is related to the `babel` and `regenerator-transform`.
Can you check and try this out ? 
https://github.com/facebook/regenerator/issues/391#issuecomment-617095947

```
Does your targets support destructuring?

    Yes -> Then they also support generators, so you can remove @babel/plugin-transform-regenerator
    No -> Then you need to add @babel/plugin-transform-destructuring
```

look your `package.json` file
and handle the case.

**Reply by burakkaraceylan** [reply]
Thank you for the answer. This is a react application created with create-react-app. It compiles fine if I don't import model-viewer as a module and just import it from cdn. Therefore, I thought the problem is with model-viewer library. I am still not sure on how to handle this with react? As I said it works fine in development but the error occurs when I try to build it with parcel. If the problem is related to babel howcome it works if I don't import model-viewer library?

**Reply by shamsail** [reply]
yeah this is the point where I'm confused because if something wrong with the model-viewer in this matter than it should not work at all. It's your project settings or dependencies that is causing a problem with the library. (I'm not expert but it seems to be a package errors) 
Can you check which `npm` version you are using?  `model-viewer` has some issues with `npm version 7` . So, may be then you can use with `npm version 6.14` . It's just my another thought that can cause problem in package

**Reply by burakkaraceylan** [reply]
Hello! Sorry for the late answer we had a pretty long holiday here.  I've downgraded npm to  6.14 but the problem persists. 

Also, ejecting from create-react-app and adding "@babel/plugin-transform-destructuring" didn't help either. Where can I get some help? I have to deploy my project and I can't get past this problem.
